src/qt/window.cpp

```diff
@@ -1510,14 +1518,21 @@ bool wxWindowQt::QtHandleWheelEvent ( QWidget *WXUNUSED( handler ), QWheelEvent
     wxMouseEvent e( wxEVT_MOUSEWHEEL );
 #if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
     QPoint qPt = event->position().toPoint();
+    wxMouseWheelAxis wheelAxis = event->angleDelta().y() > 0
```

Comment:
Seems to be missing abs().

Also I wonder if both x and y could be non-zero. In this case maybe it should send two events similar to https://github.com/wxWidgets/wxWidgets/issues/24516

(repeating because apparently my previous comment was not visible)

Reply:
> Seems to be missing abs().
> 
> Also I wonder if both x and y could be non-zero. In this case maybe it should send two events similar to #24516
> 

Can you please test this patch (which is, unfortunately, for Qt 5.15+ only) and tell me if it works properly for you (TIA!):

```
diff --git a/src/qt/window.cpp b/src/qt/window.cpp
index 0a5ff12172..ed6655b1af 100644
--- a/src/qt/window.cpp
+++ b/src/qt/window.cpp
@@ -1516,27 +1516,42 @@ bool wxWindowQt::QtHandleResizeEvent ( QWidget *WXUNUSED( handler ), QResizeEven
 bool wxWindowQt::QtHandleWheelEvent ( QWidget *WXUNUSED( handler ), QWheelEvent *event )
 {
     wxMouseEvent e( wxEVT_MOUSEWHEEL );
+    e.m_linesPerAction = 3;
+    e.m_wheelDelta = 120;
+    e.SetEventObject(this);
+
 #if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
-    QPoint qPt = event->position().toPoint();
-    wxMouseWheelAxis wheelAxis = event->angleDelta().y() > 0
-                               ? wxMOUSE_WHEEL_VERTICAL : wxMOUSE_WHEEL_HORIZONTAL;
-    int wheelRotation = wheelAxis == wxMOUSE_WHEEL_VERTICAL
-                      ? (event->angleDelta().y() / 8) : (event->angleDelta().x() / 8);
+    e.SetPosition( wxQtConvertPoint(event->position().toPoint()) );
+
+    bool processed_X = false,
+         processed_Y = false;
+
+    if ( event->angleDelta().y() != 0 )
+    {
+        e.m_wheelAxis = wxMOUSE_WHEEL_VERTICAL;
+        e.m_wheelRotation = event->angleDelta().y() / 8;
+
+        processed_Y = ProcessWindowEvent( e );
+    }
+
+    if ( event->angleDelta().x() != 0 )
+    {
+        e.m_wheelAxis = wxMOUSE_WHEEL_HORIZONTAL;
+        e.m_wheelRotation = event->angleDelta().x() / 8;
+
+        processed_X = ProcessWindowEvent( e );
+    }
+
+    return processed_X || processed_Y;
 #else
-    QPoint qPt = event->pos();
-    wxMouseWheelAxis wheelAxis = event->orientation() == Qt::Vertical
-                               ? wxMOUSE_WHEEL_VERTICAL : wxMOUSE_WHEEL_HORIZONTAL;
-    int wheelRotation = event->delta();
-#endif
-    e.SetPosition( wxQtConvertPoint( qPt ) );
-    e.SetEventObject(this);
+    e.SetPosition( wxQtConvertPoint(event->pos()) );
 
-    e.m_wheelAxis = wheelAxis;
-    e.m_wheelRotation = wheelRotation;
-    e.m_linesPerAction = 3;
-    e.m_wheelDelta = 120;
+    e.m_wheelAxis = event->orientation() == Qt::Vertical
+                  ? wxMOUSE_WHEEL_VERTICAL : wxMOUSE_WHEEL_HORIZONTAL;
+    e.m_wheelRotation = event->delta();
 
     return ProcessWindowEvent( e );
+#endif
 }
 
 

```

Reply:
Thanks,
Yes, it works with https://github.com/p2rkw/xf86-input-mtrack, athrough I was unable to receive both non-zero x and y in `angleDelta()` at the same time.


Mouse wheel events come through correctly with same values as wxGTK.

In `scroll` sample, horizontal scrolling works in inverse direction compared to wxGTK but that's a different issue.